src/conn/conn_oligarch.c

```diff
@@ -862,24 +950,26 @@ __wti_disagg_conn_config(WT_SESSION_IMPL *session, const char **cfg, bool reconf
     /* Set the role. */
     WT_ERR(__wt_config_gets(session, cfg, "disaggregated.role", &cval));
     if (cval.len == 0)
-        conn->oligarch_manager.leader = false;
+        conn->oligarch_manager.leader = leader = false;
     else {
         if (WT_CONFIG_LIT_MATCH("follower", cval))
-            conn->oligarch_manager.leader = false;
+            conn->oligarch_manager.leader = leader = false;
         else if (WT_CONFIG_LIT_MATCH("leader", cval))
-            conn->oligarch_manager.leader = true;
+            conn->oligarch_manager.leader = leader = true;
         else
             WT_ERR_MSG(session, EINVAL, "Invalid node role");
 
         /* Follower step-up. */
-        if (reconfig && !was_leader && conn->oligarch_manager.leader) {
+        if (reconfig && !was_leader && leader) {
             /*
              * Note that we should have picked up a new checkpoint ID above. Now that we are the new
              * leader, we need to begin the next checkpoint.
              */
             if (next_checkpoint_id == 0)
                 WT_ACQUIRE_READ(
                   next_checkpoint_id, conn->disaggregated_storage.global_checkpoint_id);
+            if (next_checkpoint_id == 0)
```

Comment:
Either a comment or some `#define`s for the 0/1 numbers would be welcome here, the 0/1 feels a bit too magical.

Reply:
Sure, I will add `WT_DISAGG_CHECKPOINT_ID_NONE` and `WT_DISAGG_CHECKPOINT_ID_FIRST`.